hs_core/models.py

```diff
@@ -1750,70 +1766,494 @@ class Meta:
         unique_together = ("content_type", "object_id")
 
 
-def get_path(instance, filename):
+# TODO: change to get_resourcefile_path
+def get_path(instance, filename, folder=None):
     """
-    Dynamically determine storage path for a FileField based upon whether resource is federated
+    Get a path from a ResourceFile, filename, and folder
 
-    :param instance: instance of ResourceFile containing the FileField;
-    :param filename: the filename to be used; derived from upload data
+    :param instance: instance of ResourceFile to use
+    :param filename: file name to use (without folder)
+    :param folder: can override folder for ResourceFile instance
 
-    The instance points to the Resource record, which contains the federation path.
+    The filename is only a single name. This routine converts it to an absolute
+    path that can be federated or local.  The instance points to the Resource record,
+    which contains the federation path. The folder in the instance will be used unless
+    overridden.
+
+    Note: this does not change the default behavior.
+    Thus it can be used to compute a new path for file that
+    one wishes to move.
     """
-    # retrieve federation path -- if any -- from Resource object containing FileField
-    if instance.content_object.resource_federation_path:
-        # federated iRODS storage
-        prefix = os.path.join(instance.content_object.resource_federation_path,
-                              instance.content_object.short_id, 'data', 'contents')
-    else:
-        # local iRODS storage
-        prefix = os.path.join(instance.content_object.short_id, 'data', 'contents')
+    if not folder:
+        folder = instance.file_folder
+    return get_resource_path(instance.resource, filename, folder)
```

Comment:
@alvacouch Should this get_path() be combined with get_resource_path() below? They sound to me like doing the same thing. Also, I think get_resource_file_path() as the method name is more accurate.